src/webgpu/api/validation/capability_checks/features/texture_formats.spec.ts

```diff
@@ -161,15 +162,15 @@ g.test('canvas_configuration')
       usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
     };
 
-    if (enable_required_feature) {
-      t.expectValidationError(() => {
-        ctx.configure(canvasConf);
-      });
-    } else {
-      t.shouldThrow('TypeError', () => {
-        ctx.configure(canvasConf);
-      });
-    }
+    const expectedError =
+      enable_required_feature &&
+      (kCanvasTextureFormats as unknown as Array<GPUTextureFormat>).includes(format)
```

Comment:
optional nit: Is there a way that we can avoid all the casts? I assume you already tried, so maybe not.

Reply:
No we cannot (even `unknown` hop is needed), the other alternative is to do something like we already do:
https://github.com/gpuweb/cts/blob/ab266c93548d2cff1d15ac1fe19ee7ee5f466793/src/webgpu/web_platform/canvas/configure.spec.ts#L167-L174

but I think we should not bend the code, but rather bend TS types to achieve what we want.